src/mailadm/bot.py

```diff
@@ -36,16 +36,41 @@ def __init__(self, db: DB, account: deltachat.Account):
         self.account = account
         with self.db.read_connection() as conn:
             config = conn.config
-            self.admingrpid = config.admingrpid
-            self.admingroup = account.get_chat_by_id(int(self.admingrpid))
+            self.admingrpid = int(config.admingrpid)
+            self.admingroup = account.get_chat_by_id(self.admingrpid)
 
     @account_hookimpl
     def ac_incoming_message(self, message: deltachat.Message):
         arguments = message.text.split(" ")
-        print("process_incoming command:", arguments)
-        if not self.check_privileges(message):
-            message.create_chat()
-            message.chat.send_text("Sorry, I only take commands from the admin group.")
+        print("process_incoming message:", message.text)
+        if not self.is_admin_group_message(message):
+            chat = message.create_chat()
+            if chat.is_group():
+                if message.get_sender_contact() not in self.admingroup.get_contacts():
+                    chat.send_text("Sorry, you can not contact me in a group chat. Please use a 1:1 chat.")
+                    chat.remove_contact(self.account.get_self_contact())   # leave group
+                elif message.quote:
+                    if message.quote.get_sender_contact().addr == self.account.get_config("addr"):
+                        recipient = chat.get_name()
+                        chat = self.account.create_chat(recipient)
+                        message.set_override_sender_name(self.account.get_config("addr"))
+                        chat.send_msg(message)
+                # else ignore message, it's just admins discussing what to do with the request
+            elif message.text[0] == "/":
+                chat.send_text("Sorry, I only take commands from the admin group.")
+            else:
+                name = message.get_sender_contact().addr
+                admins = self.admingroup.get_contacts()
+                admins.remove(self.account.get_self_contact())
+                supportgroup = None
+                for chat in self.account.get_chats():
+                    if chat.get_name() == name:
+                        if chat.is_group():
+                            supportgroup = chat
+                if not supportgroup:
```

Comment:
if you break after finding, then you can do "else: " (the wonderful for-else construct :)

Reply:
I always forget whether the `else` block is executed after `break` or not x.x
but maybe now I can remember with this very good use case of for/else loops